jetbrains-core/tst/software/aws/toolkits/jetbrains/services/ecs/EcsClusterNodeTest.kt

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecs
+
+import com.intellij.testFramework.ProjectRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.ecs.model.Service
+import software.aws.toolkits.jetbrains.core.MockResourceCache
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerEmptyNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerErrorNode
+import software.aws.toolkits.jetbrains.services.ecs.resources.EcsResources
+import java.util.concurrent.CompletableFuture
+
+class EcsClusterNodeTest {
+
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    private val clusterArn = "arn:aws:ecs:us-west-2:1234567890:cluster/clusterName"
+
+    @Before
+    fun setUp() {
+        resourceCache().clear()
+    }
+
+    @Test
+    fun arnIsParsedIntoGoodName() {
+        val node = aEcsClusterNode(clusterArn)
+
+        assertThat(node.displayName()).isEqualTo("clusterName")
+    }
+
+    @Test
+    fun invalidArnFormatDoesntBreakDisplay() {
+        val badClusterArn = "arn:aws:ecs:us-west-2:1234567890:clusterArnIsIncorrect/clusterName"
+        val node = aEcsClusterNode(badClusterArn)
+
+        assertThat(node.displayName()).isEqualTo(badClusterArn)
+    }
+
+    @Test
+    fun failedCallShowsErrorNode() {
+        val node = aEcsClusterNode()
+
+        resourceCache().addEntry(
+            EcsResources.listServiceArns(clusterArn),
+            CompletableFuture.failedFuture(RuntimeException("Simulated error"))
```

Comment:
I wonder what that's going to do when we baseline on Java 11

Reply:
Its gone now, switched to using a method from Java SDK. Dont tell them